### PR TITLE
Validate pyproject.toml

### PR DESCRIPTION
**Pull request by cclauss**
https://github.com/tox-dev/pyproject-fmt
https://github.com/abravalheri/validate-pyproject

Also, `Avoid running tests that require creds on PRs on forks` by adding the changes from #563 to get CI ✅ 
* #563

### CODE REVIEW

**Review by mfisher87 — COMMENTED** [review]
This is maybe my personal preference only, but I don't like what these formatters have done. I feel the project table should be at the top, and I feel that merging the tables together is significantly less readable than previously. I've never seen a pyproject.toml written this way before, and I've seen a lot!

That said, I'm totally in favor of a validator!

File: .pre-commit-config.yaml
Comment (cclauss) [inline .pre-commit-config.yaml:24]:
```suggestion
```

**Review by mfisher87 — APPROVED** [review]
:rocket: LGTM!

**Comment by cclauss** [comment]
> we need to figure out what's going on with Travis 😱

Let's land #559 so that we can isolate those failures.

**Comment by mfisher87** [comment]
Travis failures are expected and explained by #272 

Sorry for the delay, let's roll :)

**Comment by mfisher87** [comment]
Wait... shoot, I can't bypass the check, so I can't merge.

**Comment by mfisher87** [comment]
Looks like I _can_ bypass and merge when CodeCov is failing. Whew! But I won't merge until we have an approver on #563 -- don't want to circumvent any systems in place :)

File: .travis.yml
Comment (JessicaS11) [inline .travis.yml:30]: it would be helpful for future PRs to not include changes that are in another PR (#563)

Comment (cclauss) [reply .travis.yml:30]: Yes but… It is quite useful when CI is already failing when the PR is created.  PRs that fail are almost never reviewed and merged.

**Comment by mfisher87** [comment]
:rocket: awesome idea!

**Pull request merged**
